packages/url-loader/src/plugins/cropping.ts

```diff
@@ -31,94 +36,119 @@ export function normalizeNumberParameter(param: number | string | undefined) {
+    const hasDefinedDimensions = height || width;
+    const hasValidAspectRatio = aspectRatio && cropsAspectRatio.includes(crop);
 
-  let transformationString = '';
+    // Only apply a crop if we're defining some type of dimension attribute
+    // where the crop would make sense
 
-  // Only apply a crop if we're defining some type of dimension attribute
-  // where the crop would make sense
-
-  if ( crop && ( hasDefinedDimensions || hasValidAspectRatio ) ) {
-    transformationString = `c_${crop}`;
-  }
+    if ( crop && ( hasDefinedDimensions || hasValidAspectRatio ) ) {
+      transformations.push(`c_${crop}`);
+    }
 
-  // Aspect Ratio requires a crop mode to be applied so we want to make
-  // sure a valid one is included
+    // Aspect Ratio requires a crop mode to be applied so we want to make
+    // sure a valid one is included
 
-  if ( hasValidAspectRatio ) {
-    transformationString = `${transformationString},ar_${aspectRatio}`;
-  }
+    if ( hasValidAspectRatio ) {
+      transformations.push(`ar_${aspectRatio}`);
+    }
 
-  if ( width ) {
-    transformationString = `${transformationString},w_${width}`;
-  }
+    if ( width ) {
+      transformations.push(`w_${width}`);
+    }
 
-  // Gravity of auto only applies to certain crop types otherewise
-  // errors, so default to auto only when crop matches type
+    // Some crop types don't need a height and will resize based
+    // on the aspect ratio
 
-  if ( !options.gravity && cropsGravityAuto.includes(crop) ) {
-    options.gravity = 'auto';
-  }
+    if ( !['limit'].includes(crop) && typeof height === 'number' ) {
+      transformations.push(`h_${height}`);
+    }
 
-  // Some crop types don't need a height and will resize based
-  // on the aspect ratio
+    // Gravity of auto only applies to certain crop types otherewise
+    // errors, so default to auto only when crop matches type
 
-  if ( !['limit'].includes(crop) && typeof height === 'number' ) {
-    transformationString = `${transformationString},h_${height}`;
-  }
+    if ( !gravity && cropsGravityAuto.includes(crop) ) {
+      gravity = 'auto';
+    }
 
-  // If we have gravity, apply it, but check that the gravity passed
-  // in doesn't conflict with the crop mode
+    // If we have gravity, apply it, but check that the gravity passed
+    // in doesn't conflict with the crop mode
 
-  if ( options.gravity ) {
-    if ( options.gravity === 'auto' && !cropsGravityAuto.includes(crop) ) {
-      console.warn(`Auto gravity can only be used with crop modes: ${cropsGravityAuto.join(', ')}. Not applying gravity.`);
-    } else {
-      transformationString = `${transformationString},g_${options.gravity}`;
+    if ( gravity ) {
+      if ( gravity === 'auto' && !cropsGravityAuto.includes(crop) ) {
+        console.warn(`Auto gravity can only be used with crop modes: ${cropsGravityAuto.join(', ')}. Not applying gravity.`);
+      } else {
+        transformations.push(`g_${gravity}`);gravity = gravity;
```

Comment:
question: is this formatting intentional? Like:
`transformations.push(`g_${gravity}`);gravity = gravity;`? Shouldn't it be in a separate lines?

Reply:
oops thanks good catch

Reply:
im also having a hard time understanding why i added gravity = gravity 🙃